src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = newNode;
+            tail = head;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> currentNode = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(null, value, null);
+            unlinkNode(currentNode.prev, newNode, currentNode, "add");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isIndexInInvalidRange(index)) {
+            throw new IndexOutOfBoundsException("Index" + index + "is out of bounds!");
+        }
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (isIndexInInvalidRange(index)) {
+            throw new IndexOutOfBoundsException("Index" + index + "is out of bounds!");
+        }
+        Node<T> currentNode = getNodeByIndex(index);
+        T oldNodeValue = currentNode.value;
+        currentNode.value = value;
+        return oldNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (isIndexInInvalidRange(index)) {
+            throw new IndexOutOfBoundsException("Index" + index + "is out of bounds!");
+        }
+        Node<T> currentNode = getNodeByIndex(index);
+        unlinkNode(currentNode.prev, currentNode, currentNode.next, "remove");
+        return currentNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if ((object == null && currentNode.value == null)
+                    || (object != null && object.equals(currentNode.value))) {
```

Comment:
```suggestion
            if (currentNode.item == object || currentNode.item != null && currentNode.item.equals(object)) {
```

Reply:
Checkstyle does not approve line breaks without a logical operator.